server/database/models/teamModel.ts

```diff
@@ -0,0 +1,20 @@
+import mongoose, { Schema, Model, Types } from 'mongoose';
+import { Team } from '../../../shared/types/Tournament';
+import User from '../../../shared/types/User';
+
+const UserSchema = new Schema({
+  microsoftId: String,
+  alias: String,
+});
+
+const TeamSchema = new Schema <TeamDocument, TeamModel>({
+  name: String,
+  members: [UserSchema],
+});
+
+export interface TeamDocument extends Team, mongoose.Document {
+  members: Types.Array<User>;
```

Comment:
Why are you redefining members with a different type here?

Reply:
If we would use this interface we could call some mongoose specific array methods on team.members  which  are not available in normal ES Array type.
[https://docs.mongodb.com/manual/reference/operator/update-array/](url)

Reply:
That's exactly the thing that I think you should not do in business code. That's what I meant by "poisoning" the whole codebase with infrastructural code.

IMO the business logic should not depend on any library that is related to the infrastructure. Let's say in a few months we decide that we want to switch to a different DB engine. With such a dependency you'd have to rewrite the whole application instead of rewriting just a small DB compatibility layer.

Reply:
Yes, we can talk about it, just tell us/me what time are you available soon.

Reply:
Yes, we can talk about it, just tell us/me what time are you available soon.

Reply:
Lets make it a part of the monday meeting